README.md

```diff
@@ -34,111 +23,164 @@ Automate the process of extracting documentation URLs and adding them as sources
 - **Bulk URL Loading**: Add multiple URLs to NotebookLM automatically
 - **Error Handling**: Comprehensive error messages and recovery options
 - **YouTube & Website Support**: Handles both content types
+- **Smart Version Detection** The script automatically detects version numbers in URLs and handles them intelligently:
+
+    | URL Format | `--versions` Flag | Behavior |
+    |------------|------------------|----------|
+    | `https://docs.example.com/product/3.2` | Not specified | Uses detected version `3.2` |
+    | `https://docs.example.com/product/3.2` | `--versions 2.21,latest` | Ignores detected `3.2`, uses specified versions |
+    | `https://docs.example.com/product` | Not specified | Uses default `latest` |
+    | `https://docs.example.com/product` | `--versions 2.21,latest` | Uses specified versions |
 
-## Quick Start
+- **Supported Version Patterns**: Includes `/latest`, `/3.2`, `/v3.2`, `/2.21.1`
+- **Consistent File Handling**: The script uses **predictable file handling** for easier workflows:
+  - **Extraction**: Always saves to `urls.txt` (unless `--toc-output` specified)
+  - **Notebook Mode**: Always reads from `urls.txt` (unless `--links-file` specified)
+  - **Resource Combination**: Always includes `CQA_res.txt` static resources (unless `--skip-cqa` is used)
+  - **No Guessing**: Clear, consistent behavior every time
+
+## Installation
 
 ### Prerequisites
 - Python 3.7+
 - A NotebookLM account
 
-### Installation
+### Procedure
+
+1. OPTIONAL: If the project root is different from the script directory `add_links_notebook`, navigate to the location where the `.venv` directory is located:
+  ```bash
+  # Navigate to project root (where .venv is located) 
+  cd ../add_links_notebook
```

Comment:
This step indicates that the user already has a virtual environment created. if this is the case, then they need not to run step2. Maybe, we can merge the first and second steps into one for better clarity:
1. Based on the virtual environment availability, perform one of the following steps:
- If the virtual environment is available, navigate to the location where the `.venv` directory is placed. 
- If the virtual environment is not available, create it using the following command